scarb/src/ops/proc_macro_server/mod.rs

```diff
@@ -56,10 +68,14 @@ fn handle_requests(receiver: Receiver<RpcRequest>, sender: Sender<RpcResponse>)
     }
 }
 
-fn route_request(request: RpcRequest) -> Result<Value> {
-    #[allow(clippy::match_single_binding)]
+fn route_request(proc_macros: Arc<ProcMacroHost>, request: RpcRequest) -> Result<Value> {
     match request.method.as_str() {
-        //TODO add method handlers
+        DefinedMacros::METHOD => run_handler::<DefinedMacros>(proc_macros.clone(), request.value),
         _ => Err(anyhow!("method not found")),
     }
 }
+
+fn run_handler<M: Handler>(proc_macros: Arc<ProcMacroHost>, value: Value) -> Result<Value> {
```

Comment:
[nit]
```suggestion
fn handle<M: Handler>(proc_macros: Arc<ProcMacroHost>, value: Value) -> Result<Value> {
```

Reply:
we already have `handle_requests()` and `Handler::handle()`, naming next function `handle` will make it less readable imo, now it is clear that we will execute `Handler::handle()`